### PR TITLE
Upgrade envoy + crashfunding + improve debuggability

### PR DESCRIPTION
**Pull request by andraxylia**
This PR changes the proxy sha to a newer version and adds code to dump the logs for segmentation fault. It also adds logs for previous pod, if the pod crashed.

The envoy crash is random, so this will do crashfunding, it forces everybody to run with the new envoy so that we can repro the crash. I was not able to repro after 10+ local runs and 5 on the automated tests.

### CODE REVIEW

**Comment by andraxylia** [comment]
retest ci/circleci: e2e-pilot

File: pilot/test/integration/access.go [hunk 1]
```diff
@@ -87,11 +87,11 @@ func (a *accessLogs) check(infra *infra) error {
 				logs := util.FetchLogs(client, pod, ns, container)
 
 				if strings.Contains(logs, "segmentation fault") {
-					return fmt.Errorf("segmentation fault %s", pod)
+					return fmt.Errorf("segmentation fault %s log: %s", pod, logs)
```
Comment (costinm) [inline pilot/test/integration/access.go]: Good enough for now - but in a separate PR maybe just dump the log to a file - regardless of result, just like the e2e tests are doing. We want to improve consistency - eventually all this code should be moved to common (or maybe we can reuse the same code as the e2e framework ).

But one step at a time, let's get this in first.

**Comment by rshriram** [comment]
Can we do a couple of things? redirect all test output to a file when running only in circleci (not in prow). This will allow us to browse the file in the web instead of downloading it to the local machine and grepping through it.

and dump the segfault in a separate file?

Either way, pilot-e2e fails for this PR.. not sure if its segfault or not

**Comment by rshriram** [comment]
What I hate about the current status is that there is a lot of output in the console, and we have to grep through everything to find if there was a segfault or not. Can we atleast print the "segfault" message on stderr instead of stdout ? That would allow us to have these segfault indicators in the circleci console

**Comment by kyessenov** [comment]
It does exactly that. We stopped collecting logs to see stack traces which
makes this debugging complicated

**Comment by kyessenov** [comment]
And the switch from stderr was due to the new log framework I think.

**Comment by ldemailly** [comment]
that's nice, in parallel, can circleci/minikube gurus figure out how to save coredumps ?

**Comment by andraxylia** [comment]
We figured out how to copy the core files, need to test it and will submit it soon.

File: pilot/test/integration/access.go [hunk 2]
```diff
@@ -87,11 +87,12 @@ func (a *accessLogs) check(infra *infra) error {
 				logs := util.FetchLogs(client, pod, ns, container)
 
 				if strings.Contains(logs, "segmentation fault") {
-					return fmt.Errorf("segmentation fault %s", pod)
+					util.CopyCoreFiles(pod, ns, "/etc/istio/proxy/core*", infra.coreFilesDir)
```
Comment (ldemailly) [inline pilot/test/integration/access.go]: can you save logs too ?

File: pilot/test/util/kubernetes.go [hunk 3]
```diff
@@ -105,6 +105,14 @@ func describeNotReadyPods(items []v1.Pod, kubeconfig, ns string) {
 	}
 }
 
+func CopyCoreFiles(pod, ns, source, dest string) {
+	// kubectl cp <some-namespace>/<some-pod>:/tmp/foo /tmp/bar
+	cmd := fmt.Sprintf("kubectl cp %s/%s:%s %s",
```
Comment (ldemailly) [inline pilot/test/util/kubernetes.go]: probably needs -c istio-proxy ?

File: pilot/test/integration/access.go (hunk 2 above)
Comment (costinm) [reply pilot/test/integration/access.go]: In a separate PR - no need to do everything in a mega PR, we should get this out sooner.

File: pilot/test/util/kubernetes.go (hunk 3 above)
Comment (costinm) [reply pilot/test/util/kubernetes.go]: I think 'cp' doesn't need -c, there is a single filesystem namespace for all pods.

**Comment by costinm** [comment]
lint errors now. Yet the pilot test is passing !

File: pilot/test/util/kubernetes.go (hunk 3 above)
Comment (ldemailly) [reply pilot/test/util/kubernetes.go]: I'm basing this on the help string:
```
expected 'cp <file-spec-src> <file-spec-dest> [-c container]'.
```

File: pilot/test/integration/access.go (hunk 2 above)
Comment (ldemailly) [reply pilot/test/integration/access.go]: the highest priority issue is actually that we aren't getting the actual envoy input, if we did we probably wouldn't need the core dump. we should get both but we should first get the actual output from envoy

Comment (andraxylia) [reply pilot/test/integration/access.go]: Not sure if there is any actual output that would show more than that.

**Comment by andraxylia** [comment]
Fixed the lint and this will trigger another re-run.

File: pilot/test/integration/access.go (hunk 2 above)
Comment (ldemailly) [reply pilot/test/integration/access.go]: `strings.Contains(logs,"segmentation fault"` why not print all of logs when there is such segmentation fault

**Comment by andraxylia** [comment]
Pilot is passing again. I think we just need to merge this.

**Pull request merged**